File: python/algorithm/everything_algorithm/algorithm_study/programmers/level2/이진변환반복하기/이진변환반복하기.py
def solution(s):
    zero_remove_count = 0
    count = 0

    while s != '1':

        st = ''
        for number in s:
            if number == '1':
                st += '1'
            else:
                zero_remove_count += 1

        count += 1
        decimal = len(st)
        s = bin(decimal)[2:]

    return [count, zero_remove_count]

print(solution("110010101001"))
# [3,8]

print(solution("01110"))
# [3,3]

print(solution("1111111"))
# [4,1]

# https://school.programmers.co.kr/learn/courses/30/lessons/70129

File: JavaScript/algorithm/src/algorithm/이진변환반복하기/이진변환반복하기.js
function solution(s) {
  let count = 0;
  let zero_remove_count = 0;

  while (s !== "1") {
    count++;

    let number = "";
    for (const num of s) {
      if (num === "1") {
        number += num;
      }
    }

    zero_remove_count += s.length - number.length;
    s = number.length.toString(2);
  }

  return [count, zero_remove_count];
}

console.log(solution("110010101001"));
// [3,8]

console.log(solution("01110"));
// [3,3]

console.log(solution("1111111"));
// [4,1]

// 공식 풀이 연구
